src/html.ts

```diff
@@ -411,74 +439,76 @@ class HTMLRenderer {
         } else {
           extraAttr.href = node.text;
         }
-        this.renderTag("a", node, extraAttr);
-        this.out(node.text);
-        this.renderCloseTag("a");
+        result += this.renderTag("a", node, extraAttr);
+        result += this.out(node.text);
+        result += this.renderCloseTag("a");
         break;
 
       case "strong":
-        this.inTags("strong", node, 0);
+        result += this.inTags("strong", node, 0);
         break;
 
       case "emph":
-        this.inTags("em", node, 0);
+        result += this.inTags("em", node, 0);
         break;
 
       case "span":
-        this.inTags("span", node, 0);
+        result += this.inTags("span", node, 0);
         break;
 
       case "mark":
-        this.inTags("mark", node, 0);
+        result += this.inTags("mark", node, 0);
         break;
 
       case "insert":
-        this.inTags("ins", node, 0);
+        result += this.inTags("ins", node, 0);
         break;
 
       case "delete":
-        this.inTags("del", node, 0);
+        result += this.inTags("del", node, 0);
         break;
 
       case "superscript":
-        this.inTags("sup", node, 0);
+        result += this.inTags("sup", node, 0);
         break;
 
       case "subscript":
-        this.inTags("sub", node, 0);
+        result += this.inTags("sub", node, 0);
         break;
 
       default:
     }
+    return result
```

Comment:
Instead of doing `result +=`, then `break`, then `return result`, we could simply `return` the result immediately at the end of each case. (And then the `break` wouldn't be needed.) What do you think?

Reply:
In this case we'd need an `else` for `caption` and any other case that doesn't always modify `result`.

Reply:
I'd rather leave this for a follow-up up. I am entertaining thought of rewriting this switch using a visitor device from #4